tests/migrations/test_runner_individual.py

```diff
@@ -0,0 +1,235 @@
+    )
+    runner.run_migration(
+        MigrationKey(
+            MigrationGroup.TRANSACTIONS,
+            "0002_transactions_onpremise_fix_orderby_and_partitionby",
+        ),
+        force=True,
+    )
+
+    assert get_sampling_key() == "cityHash64(span_id)"
+
+    assert connection.execute("SELECT count(*) FROM transactions_local;") == [(5,)]
+
+
+def generate_transactions() -> None:
+    from datetime import datetime
+
+    table_writer = get_writable_storage(StorageKey.TRANSACTIONS).get_table_writer()
+
+    rows = []
+
+    for i in range(5):
+        raw_transaction = get_raw_transaction()
+        # Older versions of this table did not have measurements
+        del raw_transaction["data"]["measurements"]
+        del raw_transaction["data"]["breakdowns"]
+
+        processed = (
+            table_writer.get_stream_loader()
+            .get_processor()
+            .process_message(
+                (2, "insert", raw_transaction),
+                KafkaMessageMetadata(0, 0, datetime.utcnow()),
+            )
+        )
+        rows.extend(processed.rows)
+
+    BatchWriterEncoderWrapper(
+        table_writer.get_batch_writer(metrics=DummyMetricsBackend(strict=True)),
+        JSONRowEncoder(),
+    ).write(rows)
+
+
+def test_groupedmessages_compatibility() -> None:
+
+    # these three commands are repeated
+    cluster = get_cluster(StorageSetKey.EVENTS)
+    database = cluster.get_database()
+    connection = cluster.get_query_connection(ClickhouseClientSettings.MIGRATE)
+
+    # Create old style table witihout project ID
+    connection.execute(
+        """
+        CREATE TABLE groupedmessage_local (`offset` UInt64, `record_deleted` UInt8,
+        `id` UInt64, `status` Nullable(UInt8), `last_seen` Nullable(DateTime),
+        `first_seen` Nullable(DateTime), `active_at` Nullable(DateTime),
+        `first_release_id` Nullable(UInt64)) ENGINE = ReplacingMergeTree(offset)
+        ORDER BY id SAMPLE BY id SETTINGS index_granularity = 8192
+        """
+    )
+
+    migration_id = "0010_groupedmessages_onpremise_compatibility"
+
+    # Runner() instantiation, run_migration(), update status
+    runner = Runner()
+    runner.run_migration(MigrationKey(MigrationGroup.SYSTEM, "0001_migrations"))
+    events_migrations = get_group_loader(MigrationGroup.EVENTS).get_migrations()
+
+    # Mark prior migrations complete
+    for migration in events_migrations[: (events_migrations.index(migration_id))]:
+        runner._update_migration_status(
+            MigrationKey(MigrationGroup.EVENTS, migration), Status.COMPLETED
+        )
+
+    runner.run_migration(
+        MigrationKey(MigrationGroup.EVENTS, migration_id), force=True,
+    )
+
+    assert connection.execute(
+        f"SELECT primary_key FROM system.tables WHERE name = 'groupedmessage_local' AND database = '{database}'"
+    ) == [("project_id, id",)]
+
+
+# provide a migration ID, group and run all the migrations that come before
```

Comment:
Please move this into a docstring inside the function. The documentation on what a method does is generally added there.
https://www.python.org/dev/peps/pep-0257/

Reply:
Added docstrings for the 2 new functions.